qabelbox/src/test/java/de/qabel/qabelbox/account/AccountManagerTest.java

```diff
@@ -85,5 +87,32 @@ public Boolean call() throws Exception {
 
         assertEquals(MockBlockServer.SIZE, quota.getSize());
         assertEquals(MockBlockServer.QUOTA, quota.getQuota());
+
+        //100Gb
+        long oldQuota = MockBlockServer.QUOTA;
+        long oldSize = MockBlockServer.SIZE;
+        MockBlockServer.QUOTA = 107374182400L;
+        MockBlockServer.SIZE = 1024L;
+        statusCodes[0] = 0;
+
+        quota = accountManager.getBoxQuota();
+
+        assertEquals(oldSize, quota.getSize());
+        assertEquals(oldQuota, quota.getQuota());
+        TestHelper.waitUntil(new Callable<Boolean>() {
+            @Override
+            public Boolean call() throws Exception {
+                return statusCodes[0] > 0;
+            }
+        }, "Error waiting for changed quota");
+        assertEquals(AccountStatusCodes.QUOTA_UPDATED, statusCodes[0]);
+        quota = accountManager.getBoxQuota();
+
+        assertEquals(MockBlockServer.SIZE, quota.getSize());
+        assertEquals(MockBlockServer.QUOTA, quota.getQuota());
+
+        assertEquals(FileUtils.byteCountToDisplaySize(quota.getQuota()), "100 GB");
+        assertEquals(FileUtils.byteCountToDisplaySize(quota.getSize()), "1 KB");
```

Comment:
with some refactoring and removal of 90% of the testcode, this could look like
```JAVA
MockBlockServer.QUOTA = 100_GB;  // arrange
MockBlockServer.SIZE = 1_KB;

accountManager.getBoxQuota();  // act

waitForQuota("100 GB", "1 KB");  // assert
```

Thats what you need to test that the quota is fetched. If it's really important that the immediate return value of getBoxQuota is the old quota, then add a single assertion for that.
Arrange, Act, Assert ... easy to read, easy debug on failure and everything you should need. (Even though this wasn't written test driven, the second rule of tdd does still make sense).

With kotlin you could even clear that mysterious `waitForQuita and do`
```KOTLIN
MockBlockServer.QUOTA = 100_GB
MockBlockServer.SIZE = 1_KB

accountManager.getBoxQuota()

assertAsync(() -> { accountManager.boxQuota.quota }, equalTo(100_GB))
```

And you can pay the price of a second test method. Without the diff-highlighting, I wouldn't even read this test.